attest/verifier/types/build.rs

```diff
@@ -1,6 +1,24 @@
 // Copyright (c) 2023 The MobileCoin Foundation
 
+use prost_build::Config;
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    prost_build::compile_protos(&["attest.proto"], &["../../api/proto"])?;
+    let mut config = Config::new();
+
+    // Note about `Digestible`. `Digestible` needs to be backwards compatible,
+    // meaning fields have to be added to the digest in the same order for all time.
+    // Deriving `Digestible` will add fields based on the declaration order.
+    // Fortunately prost orders fields based on tag number. This means, as long
+    // as someone doesn't break the prost tag numbers or names then,
+    // `Digestible` should be stable. There are tests in each of the
+    // `convert/<type_name>.rs` files that help ensure the `Digestible` field
+    // order stability.
+    for t in ["EnclaveReportDataContents", "Quote3", "Collateral"].iter() {
+        config.type_attribute(
+            t,
+            "#[derive(serde::Serialize, serde::Deserialize, Digestible, Eq)]",
+        );
+    }
+    config.compile_protos(&["attest.proto"], &["../../api/proto"])?;
```

Comment:
It looks like this is another instance of Nick didn't RTFM.

I think deriving is probably a better way to go. I like having the tests for ensuring the backwards compatibility.

Also I added serde and Eq here as they'll be needed soonish and figured it would keep the noise down on future PRs

Reply:
Woo! Super cool that this is possible.